pkg/staticnetworkconfig/generator.go

```diff
@@ -316,6 +317,66 @@ func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistence(macInterf
 	return nil
 }
 
+func (s *StaticNetworkConfigGenerator) validateInterfaceNamesExistenceYAML(macInterfaceMap models.MacInterfaceMap, networksYaml string) error {
+	interfaceNames := lo.Map(macInterfaceMap, func(m *models.MacInterfaceMapItems0, _ int) string { return m.LogicalNicName })
+
+	//var config InterfaceConfig
+	var config map[string]interface{}
+
+	// Unmarshal the JSON string into the config struct
+	err := yaml.Unmarshal([]byte(networksYaml), &config)
+	if err != nil {
+		fmt.Println("Error unmarshalling JSON:", err)
+		return err
+	}
+
+	var matched bool
+	Interfaces := config["interfaces"].([]interface{})
+
+	for _, iface := range Interfaces {
+		nic := iface.(map[interface{}]interface{})
+		interfaceName := nic["name"]
+		interfaceType := nic["type"]
+		switch interfaceType {
+		case "802-3-ethernet", "ethernet":
+			if !lo.Contains(interfaceNames, interfaceName.(string)) {
```

Comment:
Please notice https://github.com/openshift/assisted-service/pull/6715 is changing this validation.